plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -206,7 +207,16 @@ export class WinstonInstrumentation extends InstrumentationBase {
               let newTransports = Array.isArray(originalTransports)
                 ? originalTransports
                 : [];
-              const openTelemetryTransport = new OpenTelemetryTransportV3();
+              let transportOptions = {};
+              if (config.logSeverity) {
+                const winstonLevel = winstonLevelFromSeverity(
```

Comment:
(This could be a separate issue/PR, or here.) I gather that the result will confusing if the user configured their Winston logger with levels other than the npm defaults.  The level names returned by `winstonLevelFromSeverity` would result in `undefined` values in the comparison at https://github.com/winstonjs/winston-transport/blob/f4be5c72b5c8d90b0aae1690e077089116d4bb64/modern.js#L80

Perhaps, either add a comment in the README for this option that `logSeverity` only works with the default Winston levels (https://github.com/winstonjs/winston#logging-levels), or update this code to look at `args[0]?.levels` to try to cope. There is a limit to how well this could cope with weird levels. The built in npm, syslog, and cli levels could probably be handled.

I suppose the code could emit a diag.warn if `logSeverity` is given but `args[0].levels` is provided with unexpected level names.

Thoughts?

Reply:
@trentm  Yeah sound good, let me see how we can improve this to support other levels